### ISSUE TITLE
Problems in imagenet dataset

### ISSUE THREAD

**Issue by yyq027** [HanxunH/CognitiveDistillation#3]
Hi author, thanks for your work and code.
When I use imagenet subset for training models, The config i set is like below:
python extract.py --exp_path storage --exp_config configs/backdoor_exp/imagenet/poison_rate_0.100 --exp_name BadNetImageNet_rn18 --method "CD" --gamma 0.001 --beta 10

I'm running into a problem where the modeling isn't very effective.The difference in CP between poisoned and clean samples isn't significant.
I'm not sure if I'm missing some settings, if you know why, please give me some help, thank you!

**Comment by HanxunH** [comment]
Hi, thanks for your interest. Would you be able to check the mask visualization of the final mask?

**Comment by yyq027** [comment]
Thank you for your reply.
I set the mask visualization to look like this：(I've tried three trigger settings, but none of them work well.)
![Image](https://github.com/user-attachments/assets/a76c431f-c8a7-4a54-922b-4832dfb4e7e4)

![Image](https://github.com/user-attachments/assets/d6b93695-1579-472a-8e97-0e465b91728e)
The mask is not focused on triggers, so I'm wondering if there's a problem with the settings on Resnet18
The network settings follow：
optimizer:SGD lr: 0.1 weight_decay: 0.0001 momentum: 0.9
scheduler: CosineAnnealingLR( T_max=100, eta_min=1e-6)

**Comment by HanxunH** [comment]
This does seem a bit odd.

For the CD hyperparameter, I tuned it using the following criterion: the mask should be sufficiently focused on the main object in a randomly selected image. Based on what you’re showing, I don’t think the issue lies in the training hyperparameters or the CD configuration.

- Have you checked the ASR of the trigger?
- Also, could you try increasing the number of steps used for mask extraction to 150 or 200?

I’ve uploaded a [checkpoint and configuration file](https://drive.google.com/drive/u/0/folders/1KClBs6bhqMKOZhYP0VmhUiQjGDMxPDoC) for RN18 using the experimental setting you suggested. Could you try to reproduce the results on your end? Hopefully, this will offer some useful hints on the issue.

**Comment by HanxunH** [comment]
Also, did you use data augmentation during training?
